app/controllers/application_controller.rb

```diff
@@ -24,7 +24,7 @@ def confirm_change(object, message = nil)
 
   def report_errors(object)
     flash[:errors] = object.errors.full_messages
-    redirect_to :back
+    redirect_back fallback_location: 'public/404.html'
```

Comment:
That's the reason. I thought it was a generic path. 

What do you mean another method argument? Removing the `redirect_back` method?